### PR TITLE
Updated sections and example BoM design

### PR DESCRIPTION
**Pull request by ben-moss**
# Summary of Changes

1. Rewrite of overview/intro to align with the Sprint/Milestone brief
   1. Links to SAP literature on terminology
   1. Clear and simple phase names used for all 5 phases
   1. Removed some repetition in phase descriptions
1. Contents tweaks
   1. Give overview/intro before contents to set context
   1. Remove pointless entries to initial sections (e.g. _Contents_ itself)
   1.  Update to "new" (in fact the original) section headings that align with agreed phase naming
1. Output(s) sections
   1. Updated to convey Results too (system state changes aren't really outputs)
1. Examples
   1. Referenced by name to provide more inline context
   1. Lowered heading level (doesn't warrant contents entry, but useful to link to inline)
1. SAP Library structure
   1. Templates don't belong with stack files (not generated by SAP)
   1. Templates _probably_ should move out of the BoM as these are likely to be many-to-one entity relationship, but not actioned this without further discussion
   1. Individual notes reformatted into list of notes without **Note:** prefix at the start of each one
1. BoM structure
   1. Reworked to more yaml like format where structured data is more yaml parseable than in comments as previously defined
   1. Attempted to improve clarity of sections (i.e. now there really is a _list of materials_)
   1. Classified different types of materials so they can be handled appropriately without needless duplication of e.g. file permissions
   1. Illustrated the HANA BoM use case with dependent materials section to link to another BoM
   1. Templates now linked via materials entries (they are themselves part of the materials list that make up a deployment)
1. Deployment phase
   1. Simplified to high-level process
   1. Gave illustrative examples rather than near complete lists
   1. Linked directly to SAP notes that illustrate the type of configuration taking place in this step

## To Do

1. Preparation phase is too low level in places (i.e. telling user how to achieve a goal e.g. "click on link" rather than what the goal is e.g. "upload media to storage account").

### CODE REVIEW

File: documentation/ansible/system-design-deployment.md [hunk 1]
```diff
@@ -1,42 +1,43 @@
...
+For example, an administrator may have a requirement to deploy a consistent [3-tier system landscape](https://help.sap.com/doc/saphelp_afs64/6.4/en-US/de/6b0da2f34d11d3a6510000e835363f/content.htm?no_cache=true) for S/4HANA 1909 SPS02 over the course of a few months, where Production is not required until 2-3 months after the Development system is required.
+This can be a challenge when SAP removes available software versions, or a customer’s technical SAP personnel changes over time.
+
+The process is aimed at users with some prior experience of both deploying SAP systems and with the Azure cloud platform.
+For example, users should be familiar with: _SAP Launchpad_, _SAP Maintenance Planner_, _SAP Download Manager_, and _Azure Portal_.
+
+The process described here consists of 3 distinct phases:
+
+1. **_Acquisition_** of the SAP installation media, configuration files and tools;
+1. **_Preparation_** of the SAP media library, and generation the files required for automated deployments;
```
Comment (robfallows) [inline documentation/ansible/system-design-deployment.md:14]:
```suggestion
1. **_Preparation_** of the SAP media library, and generation of the files required for automated deployments;
```

**Review by robfallows — APPROVED** [review]
One small typo fix suggested, otherwise LGTM 👍

**Review by centiqmurphy — APPROVED** [review]
LGTM Giving the examples headers works much better, one suggestion regarding the wrong example link.

File: documentation/ansible/system-design-deployment.md [hunk 2]
```diff
@@ -89,8 +90,9 @@ _**Note:** The Preparation and Deployment stages will be independent of each oth
 
 ### Phase 2 Process
 
-1. Upload SAP Media from workstation to SAP Library. This process will create a repository of archive files, tools and Stack files to be used with deploying systems. See [Examples 1](#phase-2-examples) for file structure.
-1. Upload the downloaded media and stack files to the sapbits container in the Storage Account for the SAP Library, using the directory structure shown in [Examples 1](#phase-2-examples).
+1. Upload SAP Media from workstation to SAP Library.
+   This process will create a repository of archive files, tools and Stack files to be used when deploying systems (see [Example SAP Library file structure](#example-sap-library-file-structure).
+1. Upload the downloaded media and stack files to the sapbits container in the Storage Account for the SAP Library, using the directory structure shown in the [Example Bill of Materials (BoM) file](#example-bill-of-materials-bom-file).
```
Comment (centiqmurphy) [inline documentation/ansible/system-design-deployment.md:95]: I think this should be "Shown in the [Example SAP Library file structure]"

Comment (centiqmurphy) [inline documentation/ansible/system-design-deployment.md:95]:
```suggestion
1. Upload the downloaded media and stack files to the sapbits container in the Storage Account for the SAP Library, using the directory structure shown in the [Example SAP Library file structure](#example-sap-library-file-structure).
```

File: documentation/ansible/system-design-deployment.md [hunk 3]
```diff
@@ -103,7 +105,7 @@ _**Note:** The Preparation and Deployment stages will be independent of each oth
 1. Select all unattended installation template files (`*.j2`).
 1. Click Advanced to show the advanced options, and enter “templates” for the Upload Directory.
 1. Create the BoM file and upload it into SAP Library.
-1. Populate BoM with required inputs show in [Examples 2](#phase-2-examples)
+1. Populate BoM with required inputs shown in the [Example Bill of Materials (BoM) file](#example-bill-of-materials-bom-file).
```
Comment (centiqmurphy) [inline documentation/ansible/system-design-deployment.md:108]:
```suggestion
```

File: documentation/ansible/system-design-deployment.md [hunk 4]
```diff
@@ -112,107 +114,95 @@ _**Note:** The Preparation and Deployment stages will be independent of each oth
...
+#### Example Bill of Materials (BoM) file
+
+File `BoMs/S4HANA_1909_v001/bom.yml`:
+
+```yaml
+---
+
+name:    'S/4HANA - 1909'
+version: 001
+target:  'ABAP PLATFORM 1909'
```
Comment (peterchoo) [inline documentation/ansible/system-design-deployment.md:165]: Might want to be careful using such generic top level variable names, potential clashes if anything else uses these

Comment (ben-moss) [reply documentation/ansible/system-design-deployment.md:165]: Great point. I'll update

Comment (ben-moss) [reply documentation/ansible/system-design-deployment.md:165]: On second, thought. I'll leave for now. They can be included as:
```
- include_vars: bom.xml
    name: bom
```

Resulting in: `bom.name`, `bom.version`, etc.

Either way, easy to change, but great point for us all to note 👍

**Pull request merged**